### PR TITLE
[MOOV-1864]: Added new Payment Request Dashboard filter controls

### PR DESCRIPTION
**Pull request by agranillonf**
Added the ability to filter the payment request dashboard by:

- Text: On Merchant's name, payment request description and title, customer email address and PISP reference.
- Currency.
- Minimum requested amount.
- Maximum requested amount.
- Tags

The API client was also updated to support these filters.

### CODE REVIEW

File: src/components/ui/PaymentRequestAmountFilter/PaymentRequestAmountFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
+﻿import FilterButton from '../FilterButton/FilterButton';
```
Comment (donalnofrixion) [inline src/components/ui/PaymentRequestAmountFilter/PaymentRequestAmountFilter.tsx:1]: I'd remove PaymentRequest from the name. We'll reuse these for Accounts Payable.

File: src/components/ui/PaymentRequestTagFilter/PaymentRequestTagFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
+﻿import FilterButton from '../FilterButton/FilterButton';
```
Comment (donalnofrixion) [inline src/components/ui/PaymentRequestTagFilter/PaymentRequestTagFilter.tsx:1]: Same here

File: src/components/ui/PaymentRequestFilterRow/PaymentRequestFilterRow.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
+﻿import DateRangePicker, { DateRange } from '../DateRangePicker/DateRangePicker';
```
Comment (donalnofrixion) [inline src/components/ui/PaymentRequestFilterRow/PaymentRequestFilterRow.tsx:1]: Maybe here too?

**Comment by donalnofrixion** [comment]
Looks awesome Axel! very cool. One small thing. Should the Tag filter have a clear 'X' similar to the amount filter?
image

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

**Comment by donalnofrixion** [comment]
Selecting pounds in the amount filter doesn't seem to work.
image

**Comment by donalnofrixion** [comment]
Should the tags be alphabetically ordered?
image

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Tags
- Tag cloud should be left aligned.
- Tag component buttons are missing the hover state.
- If the user clicks on "cancel" the tags selected should go back to the last state before opening the box. Video attached showing the problem. In the video you can see that I turned on 2 tags, clicked cancel, but when the dropdown is open again the tags are still selected, even when I didn't press on "Apply".
- Clicking outside the box should be the same as clicking "Cancel".
- Font-weight of the cancel button should be 400.

https://github.com/nofrixion/nofrixion.webcomponents/assets/127946572/ba5c88d2-a8b2-4a43-a935-002104ec22e3



Amount:
- The same cancel issue happens here.
- I made a mistake in the figma. The icon should be always the circle with the euro inside, not the filter icon.
- I find out it looks confusing if we don't put a border around the selected filter. We should do something like this 
image


Search:
- We should add an icon to reset the field. This can be an X like we have in the other ones.
image


Thanks!

**Comment by agranillonf** [comment]
> Looks awesome Axel! very cool. One small thing. Should the Tag filter have a clear 'X' similar to the amount filter? image

@donalnofrixion There's no close button on the Figma, so I didn't put it in.

**Comment by agranillonf** [comment]
> Selecting pounds in the amount filter doesn't seem to work. image

@donalnofrixion It doesn't work because the necessary changes to the API haven't been deployed to develop as of this comment.

**Comment by agranillonf** [comment]
All comments have been attended and the currency filter changes have been deployed to the DEV API.

**Review by donalnofrixion — APPROVED** [review]

**Review by Publicker — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Everything looks great! The only thing I'm seeing that I didn't notice before is that numbers don't have commas to separate thousands and millions 
image

**Review by Publicker — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]

**Comment by Publicker** [comment]
🎉 🎉 🎉

**Review by donalnofrixion — APPROVED** [review]

**Pull request merged**